tracing-subscriber/src/filter/directive.rs

```diff
@@ -0,0 +1,765 @@
+        let fieldset = meta.fields();
+        let fields = self
+            .fields
+            .iter()
+            .filter_map(
+                |field::Match {
+                     ref name,
+                     ref value,
+                 }| {
+                    if let Some(field) = fieldset.field(name) {
+                        let value = value.as_ref().cloned()?;
+                        Some(Ok((field, value)))
+                    } else {
+                        Some(Err(()))
+                    }
+                },
+            )
+            .collect::<Result<FieldMap<_>, ()>>()
+            .ok()?;
+        Some(field::CallsiteMatch {
+            fields,
+            level: self.level.clone(),
+        })
+    }
+
+    pub(super) fn make_tables(
+        directives: impl IntoIterator<Item = Directive>,
+    ) -> (Dynamics, Statics) {
+        // TODO(eliza): this could be made more efficient...
+        let (dyns, stats): (BTreeSet<Directive>, BTreeSet<Directive>) =
+            directives.into_iter().partition(Directive::is_dynamic);
+        let statics = stats
+            .into_iter()
+            .filter_map(|d| d.to_static())
+            .chain(dyns.iter().filter_map(Directive::to_static))
+            .collect();
+        (Dynamics::from_iter(dyns), statics)
+    }
+}
+
+impl Match for Directive {
+    fn cares_about(&self, meta: &Metadata) -> bool {
+        // Does this directive have a target filter, and does it match the
+        // metadata's target?
+        if let Some(ref target) = self.target.as_ref() {
+            if !meta.target().starts_with(&target[..]) {
+                return false;
+            }
+        }
+
+        // Do we have a name filter, and does it match the metadata's name?
+        // TODO(eliza): put name globbing here?
+        if let Some(ref name) = self.in_span {
+            if name != meta.name() {
+                return false;
+            }
+        }
+
+        // Does the metadata define all the fields that this directive cares about?
+        let fields = meta.fields();
+        for field in &self.fields {
+            if fields.field(&field.name).is_none() {
+                return false;
+            }
+        }
+
+        true
+    }
+
+    fn level(&self) -> &LevelFilter {
+        &self.level
+    }
+}
+
+impl FromStr for Directive {
+    type Err = ParseError;
+    fn from_str(from: &str) -> Result<Self, Self::Err> {
+        lazy_static! {
+            static ref DIRECTIVE_RE: Regex = Regex::new(
+                r"(?x)
+                ^(?P<global_level>trace|TRACE|debug|DEBUG|info|INFO|warn|WARN|error|ERROR|off|OFF|[0-5])$ |
+                ^
+                (?: # target name or span name
+                    (?P<target>[\w:]+)|(?P<span>\[[^\]]*\])
+                ){1,2}
+                (?: # level or nothing
+                    =(?P<level>trace|TRACE|debug|DEBUG|info|INFO|warn|WARN|error|ERROR|off|OFF[0-5])?
```

Comment:
One more :slightly_smiling_face: 
```suggestion
                    =(?P<level>trace|TRACE|debug|DEBUG|info|INFO|warn|WARN|error|ERROR|off|OFF|[0-5])?
```